types/three/src/animation/KeyframeTrack.d.ts

```diff
@@ -4,6 +4,18 @@ import { CubicInterpolant } from './../math/interpolants/CubicInterpolant';
 import { Interpolant } from '../math/Interpolant';
 import { InterpolationModes } from '../constants';
 
+export interface KeyframeTrackJSON<Type extends string = 'KeyframeTrack'> {
+    type: Type;
```

Comment:
Can you help me understand how the `type` property works and why it's generic in all of them?

Reply:
> Can you help me understand how the type property works and why it's generic in all of them?
Well first, i added it because i thought the type and generator in the Meta changed based on the type. but i was wrong and just kept it generic on all of them.

But it might be good for extended objects, like bone which is just Object3D with the .type as "Bone" 
I was thinking it would be better than overriding the type property, like Group

> I would really like to avoid creating a file that doesn't exist in three.js. Looks likes this is used in 5 places. Could we either just in-line it in the places it's used (and get rid of the interface altogether) or move it into a file that already exists like Object3D.d.ts?
Yeah i was thinking the same thing, i just didnt know what to do, i'll move it to `Object3D.d.ts`

I gave the meta its own interface just incase it changes, we woulnt have to change it in everything.
Since its pretty much the same, keeps from stuff being repetitive